### DISCUSSION TITLE
Input a date through a prompt window and then use it

### DISCUSSION THREAD

**Discussion by tegowai** [SilentVoid13/Templater#1125] [Help]
I have one usecase, where i (while applying one of templates) input the needed date and then use it... tp.date.now() does not suit me and i decided to do it manually

-----
insertDate.js
async function insertDate(tp) {
  const mmnt = await tp.system.prompt("Enter the date (YYYY-MM-DD):");
  tp.store.set("manualDate",mmnt);
}

module.exports = insertDate;

---------
manualDate.js
async function manualDate(tp,format) {
    const storedDate = tp.store.get("manualDate");
    if (storedDate) {
        const mmnt = moment(storedDate);
        return mmnt.format(format);
    }
}

module.exports = manualDate;

----------
this template  gives error

<% tp.user.insertDate(tp) %>

<% tp.user.manualDate("DD MMM YYYY, dddd") %>

i understand that insertDate has no time for saving the manually entered date in order to give it further to manualDate script. How can i solve this?

**Comment by AB1908** [comment]
Hmm this doesn't make much sense. That said, you could try this out: #1097

**Reply by AB1908** [reply]
Meant to link #1098 instead

**Reply by tegowai** [reply]
why doesn't make sense? I manually give yyyy-mm-dd date through the prompt - and i want to use it, parse it like i do with tp.date.now().

**Reply by AB1908** [reply]
I meant to say that I didn't follow your code. Does it actually work?

**Reply by tegowai** [reply]
> I meant to say that I didn't follow your code. Does it actually work?

no, it doesn't. What it meant to do: i give a date through prompt in one script, it must somehow store this string globally for other script, which parses it and puts wherever it's needed in my template - that's all. But it gives some error - which, by the way, i cannot even check, because haven't found a way to open console window in obsidian

**Reply by Zachatoo** [reply]
To check for errors, first open the developer tools by pressing `Ctrl-Shift-I` on Windows/Linux or `Cmd-Opt-I` on macOS, and then go to the "Console" tab.
